server/index.js

```diff
@@ -17,14 +28,37 @@ app.use(bodyParser.json());
 
 app.use(cors());
 
-app.use(express.static(__dirname + '/../public'));
-app.use('/:id', express.static(__dirname + '/../public'));
+// app.use('/:id', express.static(__dirname + '/../public'));
+
+
 
 app.use('/rooms/:id', express.static(__dirname + '/../client/dist'));
 
+app.use(express.static(__dirname + '/../public'));
+
+
 app.post('/rooms/addRoomSummary', controllers.addSummaryInfoToRoom);
 
-app.get('/rooms/:id/summary', controllers.getRoomSummaryInfo);
+app.get('/rooms/:id/summary', async (req, res) => {
+  let stayId = req.params.id;
+    client.get(stayId, (err, data) => {
+      if (err)
+      console.log('here with error', data);
+      if(data) {
+        res.status(200).send(data);
+      } else {
+        getRoomSummary(stayId, (summary, err) => {
+          if(summary === null) {
+            res.sendStatus(err);
+          } else {
+            client.setex(stayId, 3000, JSON.stringify(summary));
```

Comment:
Only caching for 3 seconds might not be long enough to see effect, unless you repeatedly hit same ID route